nodejs/spa/src/main/webapp/index.html

```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Keycloak Single-Page Application Example</title>
+    <script type="text/javascript" src="KC_URL/js/keycloak.js"></script>
+    <script type="text/javascript">
+        function output(content) {
+            if (typeof content === 'object') {
+                content = JSON.stringify(content, null, 2)
+            }
+            document.getElementById('output').textContent = content;
+        }
+
+        function profile() {
+            if (keycloak.idTokenParsed.name) {
+                document.getElementById('name').textContent = 'Hello ' + keycloak.idTokenParsed.name;
+            } else {
+                document.getElementById('name').textContent = 'Hello ' + keycloak.idTokenParsed.preferred_username;
+            }
```

Comment:
This can be simplified by using a ternary and string interpolation
```suggestion
            document.getElementById('name').textContent = `Hello ${keycloak.idTokenParsed.name || keycloak.idTokenParsed.preferred_username}`;
```